config/config.go

```diff
@@ -182,6 +182,7 @@ func (cfg *Config) addTransports(h host.Host) error {
 			fx.ParamTags(`group:"security"`),
 		)),
 		fx.Supply(cfg.Muxers),
+		fx.Supply(h.ID()),
```

Comment:
Why this change? Did this not work before?

Reply:
The insecure transport consume the `peer.ID`: https://github.com/libp2p/go-libp2p/blob/d8d2efafdd368bc4911c734ee284ff285fcdfbca/core/sec/insecure/insecure.go#L43

The other transports consume the private key.